src/PrestaShopBundle/Controller/Admin/ModuleController.php

```diff
@@ -314,17 +314,441 @@ public function notificationAction()
         ));
     }
 
-    public function getPreferredModulesAction()
+    public function getPreferredModulesAction(Request $request)
     {
-        $controller = new \AdminModulesControllerCore();
-        ob_start();
+        $tabModulesList = $request->request->get('tab_modules_list');
+        $back = $request->request->get('back_tab_modules_list');
 
-        $controller->ajaxProcessGetTabModulesList();
+        if ($back) {
+            $back .= '&tab_modules_open=1';
+        }
+        if ($tabModulesList) {
+            $tabModulesList = explode(',', $tabModulesList);
+            $modulesListUnsorted = $this->getModulesByInstallation($tabModulesList, $request->request->get('admin_list_from_source'));
+        }
+
+        $installed = $uninstalled = array();
+
+        foreach ($tabModulesList as $key => $value) {
+            $continue = 0;
+            foreach ($modulesListUnsorted['installed'] as $moduleInstalled) {
+                if ($moduleInstalled->name == $value) {
+                    $continue = 1;
+                    $installed[] = $moduleInstalled;
+                }
+            }
+            if ($continue) {
+                continue;
+            }
+            foreach ($modulesListUnsorted['not_installed'] as $moduleNotinstalled) {
+                if ($moduleNotinstalled->name == $value) {
+                    $uninstalled[] = $moduleNotinstalled;
+                }
+            }
+        }
+
+        $moduleListSorted = array(
+            'installed' => $installed,
+            'notInstalled' => $uninstalled
+        );
+
+        $twigParams = array(
+            'currentIndex' => '',
+            'modulesList' => $moduleListSorted
+        );
+
+        if ($request->request->has('admin_list_from_source')) {
+            $twigParams['adminListFromSource'] = $request->request->get('admin_list_from_source');
+        }
 
-        $content = ob_get_clean();
-        return new Response($content);
+        return $this->render('PrestaShopBundle:Admin/Module:tab-modules-list.html.twig', $twigParams);
     }
 
+    protected function getModulesByInstallation($tab_modules_list = null, $install_source_tracking = false)
+    {
+        $addonsProvider = $this->get('prestashop.core.admin.data_provider.addons_interface');
+        $listModulesPartners = [];
```

Comment:
```php
        $filters = new AddonListFilter();
        $filters->setOrigin(AddonListFilterOrigin::ADDONS_NATIVE)
            ->setStatus(~ AddonListFilterStatus::INSTALLED);
```

Then, unfortunately, you have to filter by author (PrestaShop) or category name (Natif)